### PR TITLE
Fix usage of errno in rt-app.c

### PR DESCRIPTION
**Pull request by qais-yousef**
Some functions return 0 on success and a nonezero error number on
failure. Others return 0 on success and -1 on failure and they set the
errno.

First include <errno.h> to correctly get the definition of errno then
don't set errno manually if the function return -1 and sets the errno
itself.

Signed-off-by: Qais Yousef <qais.yousef@arm.com>

### CODE REVIEW

**Comment by qais-yousef** [comment]
@jlelli @vingu-linaro @credp 

If any of you can spend sometime to review this that'd be appreciated. Thanks!

File: src/rt-app.c [hunk 1]
```diff
@@ -50,7 +51,6 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 #define FORKS_LIMIT		1024
 
-static int errno;
```
Comment (credp) [inline src/rt-app.c]: This looks like the right change to make - I think you're never supposed to define your own errno, rather include the header.

File: src/rt-app.c [hunk 2]
```diff
@@ -927,7 +927,6 @@ static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
 				if (ret != 0) {
 					log_critical("[%d] setpriority "
 					     "returned %d", data->ind, ret);
-					errno = ret;
```
Comment (credp) [inline src/rt-app.c]: I think we should never be setting errno directly when detecting errors from library calls so LGTM.

File: src/rt-app.c [hunk 3]
```diff
@@ -1017,8 +1015,7 @@ void *thread_body(void *arg)
 	{
 		log_notice("[%d] Locking pages in memory", data->ind);
 		ret = mlockall(MCL_CURRENT | MCL_FUTURE);
-		if (ret < 0) {
-			errno = ret;
+		if (ret != 0) {
```
Comment (credp) [inline src/rt-app.c]: This change is fine, but seems to me a matter of style as this fn can only return 0 or -1. I don't mind the change though, it is clearer that 0 is the only acceptable result.

Comment (qais-yousef) [reply src/rt-app.c]: I made this change to be consistent with the rest of the code that checks for (ret != 0)

**Pull request merged**